### PR TITLE
fiscal year feature

### PR DESCRIPTION
**Pull request by sangamsth**

### CODE REVIEW

**Review by Sagar535 — APPROVED** [review]

File: spec/nepali_calendar_spec.rb [hunk 1]
```diff
@@ -103,6 +103,51 @@
...
+      ad_date = NepaliCalendar::BsCalendar.end_of_fiscal_year_in_ad('7778')
+      expect(ad_date.year.to_i).to eq(2021)
+      expect(ad_date.month.to_i).to eq(7)
+      expect(ad_date.day.to_i).to eq(15)
+    end
+
+    it 'returns the fiscal year represented as a string' do
+      fiscal_year = NepaliCalendar::BsCalendar.fiscal_year_for_bs_date(2077, 04, 01)
+      expect(fiscal_year.to_i).to eq(7778)
+    end
```
Comment (sarojkh) [inline spec/nepali_calendar_spec.rb:139]: We want to check the border jump here. The existing test should also be backed by checking the day before and the expected fiscal year. Place it inside this `it` block/example itself.

File: spec/nepali_calendar_spec.rb [hunk 2]
```diff
@@ -103,6 +103,51 @@
...
+
+    it 'returns the fiscal year represented as a string from date object(AD)' do
+      fiscal_year = NepaliCalendar::BsCalendar.fiscal_year_for_ad_date(("20210505").to_date)
+      expect(fiscal_year.to_i).to eq(7778)
+    end
```
Comment (sarojkh) [inline spec/nepali_calendar_spec.rb:144]: Same as above. Gotta check the border jump.

**Review by sarojkh — CHANGES_REQUESTED** [review]
@sangamsth Nice job! 

I would like to request some more changes, which deviate from the original issue ticket. 

1. NepaliCalendar::AdCalendar should have its own `get_fiscal_year(ad_date)`
2. NepaliCalendar::BsCalendar should have its own `get_fiscal_year(bs_date)`
3. NepaliCalendar::AdCalendar should **also**  have its own `current_fiscal_year`
4. Introduce FiscalYear as a class. I was thinking the following structure for it. 
```ruby
class NepaliCalendar::FiscalYear

# These are string representation of year like '77', '78'
attr_accessor :start_year, :end_year

def initialize(start_year, end_year)
# TODO
end

# Should return_the_date in bs
def beginning_of_year
# TODO
end

# Should return the date in bs
def end_of_year
# TODO
end

# Should return the '7879' form of string.
def to_s
# TODO
end
```

**Comment by sarojkh** [comment]
Closing in favor of #5

**Pull request closed without merging**